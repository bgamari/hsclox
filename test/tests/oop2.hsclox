class Brioche {}
print Brioche;

class Toast {}
var toast = Toast();
print toast.jam = "grape"; // Prints "grape".
