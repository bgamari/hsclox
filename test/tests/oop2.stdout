== <script> ==
0000    1 OP_CLASS            0 'Brioche
0002    | OP_DEFINE_GLOBAL    0 'Brioche
0004    2 OP_GET_GLOBAL       0 'Brioche
0006    | OP_PRINT
0007    4 OP_CLASS            1 'Toast
0009    | OP_DEFINE_GLOBAL    1 'Toast
0011    5 OP_GET_GLOBAL       1 'Toast
0013    | OP_CALL             0
0015    | OP_DEFINE_GLOBAL    2 'toast
0017    6 OP_GET_GLOBAL       2 'toast
0019    | OP_CONSTANT         4 'grape
0021    | OP_SET_PROPERTY     3 'jam
0023    | OP_PRINT
0024    7 OP_NIL
0025    | OP_RETURN
Brioche
grape
